services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -214,6 +232,68 @@ class HakemusService(
         return hakemusRepository.save(hakemus).toHakemus()
     }
 
+    /**
+     * Create a new cable report application for an excavation announcement that has a cable report
+     * applied for.
+     *
+     * @return the new cable report application entity or null if no new application was applied for
+     */
+    private fun newJohtoselvitysAppliedFor(hakemus: HakemusEntity): HakemusEntity? {
+        val hakemusEntityData = hakemus.hakemusEntityData
+        val appliedFor =
+            when (hakemusEntityData) {
+                is KaivuilmoitusEntityData -> !hakemusEntityData.cableReportDone
+                else -> return null
+            }
+        if (!appliedFor) {
+            return null
+        }
+        logger.info("Creating a new johtoselvityshakemus for a kaivuilmoitus. id=${hakemus.id}")
+        val johtoselvityshakemusData =
+            hakemusEntityData.createAccompanyingJohtoselvityshakemusData()
+        val johtoselvityshakemus =
+            hakemus
+                .copy(
+                    id = 0,
+                    applicationType = ApplicationType.CABLE_REPORT,
+                    hakemusEntityData = johtoselvityshakemusData,
+                )
+                .apply {
+                    yhteystiedot =
+                        hakemus.yhteystiedot
+                            .mapValues { it.value.copyToHakemus(this) }
+                            .toMutableMap()
+                }
```

Comment:
Yhteystiedot can be handled in the copy, no need for a separate apply:
```suggestion
                .copy(
                    id = 0,
                    applicationType = ApplicationType.CABLE_REPORT,
                    hakemusEntityData = johtoselvityshakemusData,
                    yhteystiedot = hakemus.yhteystiedot
                        .mapValues { it.value.copyToHakemus(hakemus) }
                        .toMutableMap(),
                )
```

Reply:
Oh right, that doesn't work because `this` is the copied hakemus, not the same as `hakemus`.